src/main/scala/uk/gov/nationalarchives/tdr/api/http/Routes.scala

```diff
@@ -23,6 +25,29 @@ class Routes(val config: Config) extends Cors {
   val ttlSeconds: Int = 10
   val url: String = config.getString("auth.url")
 
+  def unmarshalEntity(entity: ResponseEntity): String = {
+    Unmarshal(entity).to[String].toString
+  }
+
+  def isNon200Status(status: StatusCode): Boolean = {
+    val code = status.intValue.toString
+    !code.startsWith("2")
+  }
+
+  def logging: Directive0 = {
+    def logNon200Response(req: HttpRequest)(res: RouteResult): Unit = {
+      res match {
+        case Complete(resp) =>
+          if (isNon200Status(resp.status)) {
```

Comment:
What about `resp.status.isFailure`? I think that covers all the cases we're interested in, and it means you don't have to inspect the status code number.